tools/integration_tests/managed_folders/view_permissions_test.go

```diff
@@ -0,0 +1,218 @@
+			if objs[0].Name() != ManagedFolder1 || objs[0].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", EmptyManagedFolder1, objs[0].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder2     -- ManagedFolder2
+			if objs[1].Name() != ManagedFolder2 || objs[1].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", EmptyManagedFolder2, objs[1].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/simulatedFolder   -- SimulatedFolder
+			if objs[2].Name() != SimulatedFolder || objs[2].IsDir() != true {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", SimulatedFolder, objs[2].Name())
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/testFile  -- File
+			if objs[3].Name() != File || objs[3].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+			return nil
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == ManagedFolder1 {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder1/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == ManagedFolder2 {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+			// testBucket/NonEmptyManagedFoldersTest/managedFolder2/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		// Check if subDirectory is empty.
+		if dir.Name() == SimulatedFolder {
+			// numberOfObjects - 1
+			if len(objs) != 1 {
+				t.Errorf("Incorrect number of objects in the directory %s expectected %d: got %d: ", dir.Name(), 1, len(objs))
+			}
+
+			// testBucket/NonEmptyManagedFoldersTest/simulatedFolder/testFile  -- File
+			if objs[0].Name() != File || objs[0].IsDir() != false {
+				t.Errorf("Listed incorrect object expectected %s: got %s: ", File, objs[3].Name())
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		t.Errorf("error walking the path : %v\n", err)
+		return
+	}
+
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
+func TestManagedFolders_FolderViewPermission(t *testing.T) {
+	ts := &managedFoldersViewPermission{}
+
+	// Run tests for mountedDirectory only if --mountedDirectory  and --testBucket flag is set.
+	if setup.AreBothMountedDirectoryAndTestBucketFlagsSet() {
+		test_setup.RunTests(t, ts)
```

Comment:
I think, for this test also, we need to fetch credentials and apply permission on bucket?

Reply:
These tests will not run with mounted directory flag as mountedDirectory flag is specially for GKE and in GKE environment we will not be able to do the same. We haven't provide any tests related to `--key-file` from start only.